### ISSUE TITLE
Questions Regarding Checkpoints, Metrics, and Reproduced Results

### ISSUE THREAD

**Issue by gongda0e** [thechargedneutron/FIction#5]
Hi @thechargedneutron,

Thank you for the nice work!
I have a few questions:

**1) Release of Trained Checkpoints**
Would it be possible to release the trained checkpoints used in your experiments?
Access to the checkpoints would be extremely helpful for reproducing the results and building upon your work.
If possible, could you kindly share them?

**2) Metric Clarification**
Regarding the interaction location prediction experiments, I have a few questions about the reported metric:
-  It seems that multiple thresholds are used for evaluating the Chamfer Distance. Did you report the value corresponding to 'eval_best_chamfer_dist'?
- Additionally, for the values reported in the paper and code, what scaling factors were applied to 'eval_pr_auc' and Chamfer Distance respectively?

For reference, here is the evaluation results from the code:

<img width="1099" alt="Image" src="https://github.com/user-attachments/assets/30446174-c8a3-4f24-b15c-4988d04aeca5" />

**3) Reproduced results**
I have conducted experiments on both interaction location prediction and pose prediction tasks.
Here are my reproduced results:

<img width="1049" alt="Image" src="https://github.com/user-attachments/assets/471b914c-0dbb-4449-8d23-9200220a264c" />

<img width="1108" alt="Image" src="https://github.com/user-attachments/assets/e707b2af-8921-4350-a1be-ac1dec7544cc" />

- For interaction location prediction, I multiplied 100 for 'eval_pr_auc' and 1000 for 'eval_best_chamfer_dist', following the scale in the paper.
Please let me know if this scaling is incorrect.

However, I observed some gaps between the paper’s reported performance and my reproduced results.
Here are the training settings I used, based on your paper and code:
- interaction location prediction: epochs 3, learning rate 5e-5
- pose prediction: epochs 6, learning rate 5e-6
- other settings were kept the same as https://github.com/thechargedneutron/FIction/blob/main/main.py#L678

Would you have any advice or additional tips to help reproduce the reported performance more closely?
Any guidance would be greatly appreciated!

**4)Number of betas in SMPL parameter**
When I run the pose anticipation experiments, I consistently see the following warning:
“You are using a SMPL model, with only 10 shape coefficients.”
Is this expected behavior, or should I modify the number of shape coefficients?

**Comment by thechargedneutron** [comment]
2) You are correct.

4) Yes, this can be safely ignored.

3) I will have to investigate, I do have the checkpoints, let me take a look at it and will release that soon too. If the released checkpoints also do not have the required performance, I will check the implementation again.

btw, were you able to get the numbers with EgoVLPv2 features?

**Comment by gongda0e** [comment]
I do not use EgoVLPv2 features. The reproduced performance is obtained either without any visual features or using the Omnivore features provided by the official dataset.

Could you clarify the scale factors used for eval_pr_auc and Chamfer distance?
Would it be correct to multiply eval_pr_auc by 100 and eval_best_chamfer_dist by 1000?

**Comment by thechargedneutron** [comment]
For PR-AUC it is correct to multiply by 100. However, for chamfer, I used the non-normalized version. You have to turn off the normalization here: https://github.com/thechargedneutron/FIction/blob/main/main.py#L496. I believe the value based on my setting was around 1350 so you need to either multiply by that number or simply run with `normalize=False`.

I attached a checkpoint for Cooking so it should match with the interaction location prediction experiment. Also, here is the PR-AUC plot that shows the attached checkpoint gives the number reported in the paper.

Link to the checkpoint: https://drive.google.com/drive/folders/1_WqYCYRThVvXq80nP2MrVvIuqwmV5yJ4?usp=sharing

<img width="601" alt="Image" src="https://github.com/user-attachments/assets/2d060856-45b3-44f1-8c71-2cf4095d420b" />

Please let me know if this works! And thanks for taking the effort to reproduce the numbers. It will be very helpful for upgrading the code. I am working towards a deadline so will be slightly slower in responding. Thanks!

**Comment by neilgautam24-tamu** [comment]
Hi @thechargedneutron @gongda0e , I have used the checkpoint that has been released & the EgoVLP2 Features with changing the normalization parameter in main.py to` False`. However, PR_AUC still is not close to the one reported in the paper. 



<img width="626" height="338" alt="Image" src="https://github.com/user-attachments/assets/8b07ac1c-acc4-4446-8895-2c281cd86f4e" />

I have also attached the results while keeping the normalization parameter to `True`

<img width="479" height="356" alt="Image" src="https://github.com/user-attachments/assets/fef18deb-fc2b-4a51-9ac3-dedfc9208991" />


Could you please point out the reasons for the inferior quantitative results? Since I have followed all the mentioned steps in this work and also confirmed the 3d bounding boxes with the LLAMA based interaction results (Basically I used custom vocabulary for this scenario to find all the interacting objects using Detic as well --> This part seemed to me kind of uncertain in terms of results but since I resolved that as well I'm wondering what might be the reason for inferior results). 

I've extracted the above results specifically on the "iiith_cooking_43_1" scenario

**Comment by CorrineQiu** [comment]
Hi @thechargedneutron @neilgautam24-tamu 

I noticed that you used EgoVLPv2 features pre-trained on Ego-Exo4D in your experiment. Could you please share where you obtained these features? It would be very helpful if you could provide some access paths or relevant resource links.

Thank you for your help, and I look forward to your reply!

**Comment by sluque1** [comment]
Hi @neilgautam24-tamu,

I am also getting similar results to yours when training and testing the model on Cooking takes (~10% PR-AUC). Were you able to reproduce the results reported in the paper? If so, how? Thank you!

**Comment by sluque1** [comment]
Hi @thechargedneutron,

Thank you for releasing the checkpoint. I notice in the configuration and by the name of the folder "2024-10-31_16/49/55_v6_FINAL_no_env_cooking_34419251_lr_5e-5_ft_180_obst_30" that this checkpoint was obtained without using the environment for training the model ("use_env = False"). You mention this checkpoint gives the number reported in the paper, however in the paper the ablation study w/o environment reports a pr-auc of 9.9% while your plot suggests a pr-auc value above 20%. Can you please clarify this?
